catalyst-gateway/tests/api_tests/utils/signed_doc.py

```diff
@@ -149,3 +175,254 @@ def build_signed_doc(
 
         signed_doc_hex = signed_doc_file.read().hex()
         return signed_doc_hex
+
+
+# ------------------- #
+# Signed Docs Factory #
+# ------------------- #
+
+
+class ProposalParameterType(IntEnum):
+    CATEGORY = 0
+    CAMPAIGN = 1
+    BRAND = 2
+
+
+# return a Proposal document which is already published to the cat-gateway and the corresponding RoleID
+@pytest.fixture
+def proposal_doc_factory(
+    rbac_chain_factory,
+    proposal_form_template_doc_factory,
+    category_parameters_doc_factory,
+    campaign_parameters_doc_factory,
+    brand_parameters_doc_factory,
+):
+    def __factory__(
+        parameter_type: ProposalParameterType,
+    ) -> tuple[SignedDocument, RoleID]:
+        param: SignedDocumentBase
+        if parameter_type == ProposalParameterType.CATEGORY:
+            param = category_parameters_doc_factory
+        elif parameter_type == ProposalParameterType.CAMPAIGN:
+            param = campaign_parameters_doc_factory
+        elif parameter_type == ProposalParameterType.BRAND:
+            param = brand_parameters_doc_factory
+        else:
+            raise Exception("Invalid parameter type for proposal document")
+
+        template: SignedDocumentBase = proposal_form_template_doc_factory
+
+        result = generic_doc_builder(
+            rbac_chain_factory(),
+            DOC_TYPE["proposal"],
+            [
+                {
+                    "id": template.metadata["id"],
```

Comment:
this is a mistake the template documents, must be referenced in the `template` metadata field. You can place them inside the `parameters` filed.
I see what you are trying to do, which totally make sense, I would modify the `generic_doc_builder` interface to something like this
```
rbac_chain: RBACChain, doc_type: str, metadata: dict, content: Any
```
so you would pass a generic `metadata` field which could be different for different document types